pkg/controller/agent/pod.go

```diff
@@ -323,3 +320,111 @@ func certificatesDir(association commonv1.Association) string {
+			{Name: "KIBANA_FLEET_HOST", Value: kbConnectionSettings.host},
+			{Name: "KIBANA_FLEET_USERNAME", Value: kbConnectionSettings.username},
+			{Name: "KIBANA_FLEET_PASSWORD", Value: kbConnectionSettings.password},
+			{Name: "KIBANA_FLEET_SETUP", Value: strconv.FormatBool(agent.Spec.KibanaRef.IsDefined())},
+		}
+
+		// don't set ca key if ca is not available
+		if kbConnectionSettings.ca != "" {
+			envVars = append(envVars, corev1.EnvVar{Name: "KIBANA_FLEET_CA", Value: kbConnectionSettings.ca})
+		}
+
+		return envVars, nil
+	}
+
+	return []corev1.EnvVar{}, nil
+}
+
+func getFleetSetupFleetEnvVars(agent agentv1alpha1.Agent, client k8s.Client) ([]corev1.EnvVar, error) {
+	fleetCfg := []corev1.EnvVar{}
+
+	if agent.Spec.KibanaRef.IsDefined() {
+		fleetCfg = append(fleetCfg, corev1.EnvVar{Name: "FLEET_ENROLL", Value: "true"})
+	}
+
+	if agent.Spec.FleetServerEnabled {
+		fleetURL, err := association.ServiceURL(
+			client,
+			types.NamespacedName{Namespace: agent.Namespace, Name: HTTPServiceName(agent.Name)},
+			agent.Spec.HTTP.Protocol(),
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		fleetCfg = append(fleetCfg, corev1.EnvVar{Name: "FLEET_CA", Value: path.Join(FleetCertsMountPath, certificates.CAFileName)})
+		fleetCfg = append(fleetCfg, corev1.EnvVar{Name: "FLEET_URL", Value: fleetURL})
+	} else if agent.Spec.FleetServerRef.IsDefined() {
+		assoc, err := association.SingleAssociationOfType(agent.GetAssociations(), commonv1.FleetServerAssociationType)
+		if err != nil {
+			return nil, err
+		}
+
+		if assoc != nil {
+			fleetCfg = append(fleetCfg, corev1.EnvVar{Name: "FLEET_CA", Value: path.Join(certificatesDir(assoc), CAFileName)})
+			fleetCfg = append(fleetCfg, corev1.EnvVar{Name: "FLEET_URL", Value: assoc.AssociationConf().GetURL()})
+		}
+	}
+	return fleetCfg, nil
+}
+
+func getFleetSetupFleetServerEnvVars(agent agentv1alpha1.Agent, client k8s.Client) ([]corev1.EnvVar, error) {
+	if !agent.Spec.FleetServerEnabled {
+		return []corev1.EnvVar{}, nil
+	}
+
+	fleetServerCfg := []corev1.EnvVar{
+		{Name: "FLEET_SERVER_ENABLE", Value: "true"},
+		{Name: "FLEET_SERVER_CERT", Value: path.Join(FleetCertsMountPath, certificates.CertFileName)},
+		{Name: "FLEET_SERVER_CERT_KEY", Value: path.Join(FleetCertsMountPath, certificates.KeyFileName)},
+	}
+
+	esExpected := len(agent.Spec.ElasticsearchRefs) > 0 && agent.Spec.ElasticsearchRefs[0].IsDefined()
+	if esExpected {
+		esConnectionSettings, err := extractConnectionSettings(agent, client, commonv1.ElasticsearchAssociationType)
+		if err != nil {
+			return nil, err
+		}
+
+		fleetServerCfg = append(fleetServerCfg, corev1.EnvVar{Name: "FLEET_SERVER_ELASTICSEARCH_HOST", Value: esConnectionSettings.host})
+		fleetServerCfg = append(fleetServerCfg, corev1.EnvVar{Name: "FLEET_SERVER_ELASTICSEARCH_USERNAME", Value: esConnectionSettings.username})
+		fleetServerCfg = append(fleetServerCfg, corev1.EnvVar{Name: "FLEET_SERVER_ELASTICSEARCH_PASSWORD", Value: esConnectionSettings.password})
```

Comment:
This means we have passwords in clear text in the Pod spec, which I think it is not great. Should we not at least mount them from a secret?